app/assets/javascripts/lib/models/chart_list.coffee

```diff
@@ -175,7 +181,23 @@ class @ChartList extends Backbone.Collection
           ignore_lock: true
         })
 
-    $.when.apply(null, @chart_requests).done((d) =>
+    # The @chart_requests are jqXHR objects returned by $.ajax().
+    # Here, we want to wait until all Ajax requests have returned, no matter
+    # their success. But Promise.all (and similar constructs like $.when)
+    # has a fast-failure feature if one of the requests fails.
+    # Hence, we have to wrap the jqXHRs in new Promises which will
+    # always return successfully - after all, we are not interested in the
+    # success of the requests anyways (since success is handled directly in
+    # callbacks to the Ajax calls).
+    Promise.all(@chart_requests.map((request) ->
```

Comment:
Can you explain why the change from `$.when` to `Promise.all`? Promises [are not available in Internet Explorer](http://caniuse.com/#feat=promises) yet, so they cannot be used unless you also add a Promises polyfill library for older browsers.